internal/services/cognitive/cognitive_account_rai_policy_resource.go

```diff
@@ -0,0 +1,396 @@
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"filter_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"block_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"severity_threshold": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForContentLevel(), false),
+					},
+					"source": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyContentSource(), false),
+					},
+				},
+			},
+		},
+
+		"custom_blocklist": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"block_enabled": {
+						Type:     pluginsdk.TypeBool,
+						Required: true,
+					},
+					"source": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyContentSource(), false),
+					},
+				},
+			},
+		},
+
+		"mode": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(raipolicies.PossibleValuesForRaiPolicyMode(), false),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"type": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ModelObject() interface{} {
+	return &AccountRaiPolicyResourceModel{}
+}
+
+func (r CognitiveAccountRaiPolicyResource) ResourceType() string {
+	return "azurerm_cognitive_account_rai_policy"
+}
+
+func (r CognitiveAccountRaiPolicyResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cognitive.RaiPoliciesClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model AccountRaiPolicyResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			cognitiveAccountId, err := raipolicies.ParseAccountID(model.AccountId)
+			if err != nil {
+				return err
+			}
+
+			id := raipolicies.NewRaiPolicyID(subscriptionId, cognitiveAccountId.ResourceGroupName, cognitiveAccountId.AccountName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
```

Comment:
_Some_ resources in the provider combine these into one conditional but this is inconsistent with the rest of the provider, please split these 
```suggestion
			if err != nil {
				if !response.WasNotFound(existing.HttpResponse) {
					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
				}
			}
```